Plugins/Flow.Launcher.Plugin.Explorer/Views/ActionKeywordSetting.xaml.cs

```diff
@@ -21,51 +12,74 @@ public partial class ActionKeywordSetting : Window
     {
         private SettingsViewModel settingsViewModel;
 
-        private ActionKeywordView currentActionKeyword;
+        public ActionKeywordView CurrentActionKeyword { get; set; }
 
         private List<ActionKeywordView> actionKeywordListView;
 
-        public ActionKeywordSetting(SettingsViewModel settingsViewModel, List<ActionKeywordView> actionKeywordListView, ActionKeywordView selectedActionKeyword)
-        {
-            InitializeComponent();
+        private Settings settings;
+
+        public Visibility Visible => CurrentActionKeyword.Enabled is not null ? Visibility.Visible : Visibility.Collapsed;
 
+        public ActionKeywordSetting(SettingsViewModel settingsViewModel,
+            List<ActionKeywordView> actionKeywordListView,
+            ActionKeywordView selectedActionKeyword, Settings settings)
+        {
             this.settingsViewModel = settingsViewModel;
 
-            currentActionKeyword = selectedActionKeyword;
+            this.settings = settings;
+
+            CurrentActionKeyword = selectedActionKeyword;
 
-            txtCurrentActionKeyword.Text = selectedActionKeyword.Keyword;
+            this.actionKeywordListView = actionKeywordListView;
+            
+            InitializeComponent();
 
-            this.actionKeywordListView = actionKeywordListView; 
         }
 
         private void OnConfirmButtonClick(object sender, RoutedEventArgs e)
         {
-            var newActionKeyword = txtCurrentActionKeyword.Text;
+            var newActionKeyword = TxtCurrentActionKeyword.Text;
 
             if (string.IsNullOrEmpty(newActionKeyword))
                 return;
 
-            if (newActionKeyword == currentActionKeyword.Keyword)
+            // reset to global so it does not take up an action keyword when disabled
+            if (!CurrentActionKeyword.Enabled is not null && newActionKeyword != Query.GlobalPluginWildcardSign)
```

Comment:
dont think this logic is effective, every action keyword except for content search is falling into this clause, hence update action keyword is called here and then down below in the code

Reply:
is this logic just to catch null Enabled as well as that it is false?

Reply:
something like this will work right?
if (CurrentActionKeyword.Enabled is not null && CurrentActionKeyword.Enabled == false && newActionKeyword != Query.GlobalPluginWildcardSign)

Reply:
I change it to pattern matching, which I believe is clearer. What do you think?
`if (CurrentActionKeyword is {Enabled: true, Keyword: not Query.GlobalPluginWildcardSign})`

Reply:
I am ok either way, make sure you test it though, the previous condition had the whole thing incorrectly working lol